core/base/Module.php

```diff
@@ -199,13 +209,23 @@ public function init()
      *
      * the *@app* namespace views will be looked up for view files
      *
+     * Now use Themes component to load theme from database
+     *
      * @return string
      * @see \yii\base\Module::getLayoutPath()
      */
     public function getLayoutPath()
     {
+        $this->theme = Yii::$app->themes->getTheme();
```

Comment:
The layout file must be set in the controller, as mentioned above. Also not sure about error handling, cause there should be the question:

+ what kind of errors can a theme have?